File: public/ClientSOAP.py
from zeep import Client
import sys


def calculTemps ():

    distance = sys.argv[1]
    autonomie = sys.argv[2]
    tpsRecharge = sys.argv[3]
    client = Client('http://127.0.0.1:8000/CalculTpsTrajetService.wsdl')
    result = client.service.Calcul_Temps(distance, autonomie, tpsRecharge)
    print(result)
    return result
    

File: public/ClientSOAP.js
import { createClient } from 'soap';



    // Utilisation de la fonction
     async function getCalcul(){
    res = await myObject.calcul()
    return res;
    }

    


const create = async function (args) {
    const url = 'http://127.0.0.1:8000/CalculTpsTrajetService.wsdl';

    return new Promise((resolve, reject) => {
        createClient(url, {}, function (err, client) {
            if (err) {
                console.error('Error creating SOAP client:', err);
                reject(err);
            }

            client.Calcul_Temps(args, function (err, result) {
                if (err) {
                    console.error('Error making SOAP request:', err);
                    reject(err);
                }
                resolve(result);
            });
        });
    });
};

const myObject = {
    calcul: async function () {
        var res = 5;
        var args = { distance: 1000., autonomie: 200., tpsRecharge: 0.15 };
        res = await create(args);

        return res;
    }
};


